File: employee/tests/end_to_end_test/pages/dashboard_page.py
from selenium.webdriver.common.by import By

from tests.end_to_end_test.pages.base_page import BasePage

class DashboardPage(BasePage):
    SUBMIT_NEW_EXPENSES_BUTTON = (By.ID, "show-submit")
    VIEW_MY_EXPENSES_BUTTON = (By.ID, "show-expenses")
    LOGOUT_BUTTON = (By.ID, "logout-btn")
    TITLE = (By.CSS_SELECTOR, "h1")
    EXPENSE_HEADING = (By.XPATH, "//h3[text()='My Expenses']")
    TABLE = (By.CSS_SELECTOR, "table")

    def go_to_submit_new_expense_screen(self):
        self.click(self.SUBMIT_NEW_EXPENSES_BUTTON)

    def go_to_view_my_expenses_screen(self):
        self.click(self.VIEW_MY_EXPENSES_BUTTON)

    def click_logout_button(self):
        button = self.wait_for_clickable(self.LOGOUT_BUTTON)

        # Ensure element is actually interactable
        self.driver.execute_script(
            "arguments[0].scrollIntoView({block:'center'});", button
        )

        self.wait.until(lambda d: button.is_enabled())

        button.click()

        from tests.end_to_end_test.pages.login_page import LoginPage
        return LoginPage(self.driver)

File: manager/src/test/java/com/revature/end_to_end_tests/pages/DashboardPage.java
package com.revature.end_to_end_tests.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class DashboardPage extends BasePage {

    @FindBy(id = "show-pending")
    private WebElement pendingExpensesButon;
    @FindBy(id = "show-all-expenses")
    private WebElement allExpensesButton;
    @FindBy(id = "show-reports")
    private WebElement generateReportsButton;
    @FindBy(id = "logout-btn")
    private WebElement logoutButton;
    @FindBy(id = "generate-all-expenses-report")
    private WebElement AllExpensesReport;
    @FindBy(id = "generate-pending-report")
    private WebElement PendingReport;

    private final By flashMessage = By.id("report-message");

    private By reportInputs = By.cssSelector("div.report-section input[type='text']");

    private final By reportMessage = By.id("report-message");


    public DashboardPage(WebDriver driver) {
        super(driver);
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(DEFAULT_TIMEOUT));
        PageFactory.initElements(driver, this);
    }

    // DEFINED ACTIONS
    public void goToPendingExpensesScreen() {
        wait.until(ExpectedConditions.visibilityOf(pendingExpensesButon));
        pendingExpensesButon.click();
    }

    public void goToAllExpensesScreen() {
        wait.until(ExpectedConditions.visibilityOf(allExpensesButton));
        allExpensesButton.click();
    }


    public LoginPage logout() {
        wait.until(ExpectedConditions.visibilityOf(logoutButton));
        logoutButton.click();
        return new LoginPage(this.driver);
    }

    public void goToGenerateReportsScreen() {
        wait.until(ExpectedConditions.visibilityOf(generateReportsButton));
        generateReportsButton.click();
    }

    // REPORT GENERATION
    public void generateByExpenses() {
        wait.until(ExpectedConditions.visibilityOf(AllExpensesReport));
        AllExpensesReport.click();
    }

    public void generateByPending() {
        wait.until(ExpectedConditions.visibilityOf(PendingReport));
        PendingReport.click();
    }

    public String getFlashMessage() {
        return getText(flashMessage);
    }

    public List<WebElement> getAllInputs() {
        return driver.findElements(reportInputs);
    }

    public String getMessageText() {
        return getText(reportMessage);
    }

}


